hydrolib/core/io/net/models.py

```diff
@@ -367,25 +369,167 @@ def __init__(
-            )
+        """Generate the branch offsets and the nodes.
 
+        Args:
+            mesh1d_edge_length (float): The edge length of the 1d mesh.
+            structure_chainage (Optional[List[float]], optional): A list with the structure chainages. Defaults to None.
+            max_dist_to_struc (Optional[float], optional): The maximum distance from a node to a structure. Defaults to None.
+        """
         # Generate offsets
-        self.branch_offsets = self._generate_1d_spacing(
-            anchor_pts=[0.0, self.length], mesh1d_edge_length=mesh1d_edge_length
+        self.branch_offsets = self._generate_offsets(
+            mesh1d_edge_length, structure_chainage, max_dist_to_struc
         )
         # Calculate node positions
         self.node_xy = self.interpolate(self.branch_offsets)
         # Add mask (all False)
         self.mask = np.full(self.branch_offsets.shape, False)
 
+    def _generate_offsets(
+        self,
+        mesh1d_edge_length: float,
+        structure_offsets: Optional[List[float]] = None,
+        max_dist_to_struc: Optional[float] = None,
+    ) -> np.ndarray:
+        """Generate the branch offsets.
+
+        Args:
+            mesh1d_edge_length (float): The edge length of the 1d mesh.
+            structure_offsets (Optional[List[float]], optional): A list with the structure chainages. Defaults to None.
+            max_dist_to_struc (Optional[float], optional): The maximum distance from a node to a structure. Defaults to None.
+        Raises:
+            ValueError: Raised when any of the structure offsets is smaller than zero.
+            ValueError: Raised when any of the structure offsets is greater than the branch length.
+
+        Returns:
+            np.ndarray: The generated branch offsets.
+        """
+        # Generate initial offsets
+        anchor_pts = [0.0, self.length]
+        offsets = self._generate_1d_spacing(anchor_pts, mesh1d_edge_length)
+
+        if structure_offsets is None:
+            return offsets
+
+        # Check the limits
+        if min(structure_offsets) < 0.0:
+            raise ValueError(
+                f"Distance {min(structure_offsets)} is outside the branch range (0.0 - {self.length})."
+            )
+        if max(structure_offsets) > self.length:
+            raise ValueError(
+                f"Distance {max(structure_offsets)} is outside the branch length (0.0 - {self.length})."
+            )
+
+        # Merge limits with start and end of branch
+        limits = [-1e-3] + list(sorted(structure_offsets)) + [self.length + 1e-3]
+
+        # if requested, check if the calculation point are close enough to the structures
+        if max_dist_to_struc is not None:
+            limits: List[float] = self._update_limits(max_dist_to_struc, limits)
+
+        offsets = self._add_nodes_to_segments(
+            offsets, anchor_pts, limits, mesh1d_edge_length
+        )
+
+        return offsets
+
+    def _update_limits(
```

Comment:
Technically you're not updating if you're returning a new list. Either you would actually update the provided limits, or you could rename it to something like, `_generate_extended_limits`

Reply:
Also, this could be a static method, which could then be called as `Branch._generate_extended_limits`

Reply:
Renamed it